### PR TITLE
feat: add spring datasource data property + refactor

### PR DESCRIPTION
**Pull request by aureamunoz**
I added the new property and did a refactor to go back to the first implementation with a map because we are adding more properties....I let you take a look @geoand

### CODE REVIEW

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java [hunk 1]
```diff
@@ -1,13 +1,16 @@
 package io.quarkus.spring.data.deployment;
 
+import static java.util.stream.Collectors.*;
```
Comment (geoand) [inline extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java:3]: No start imports please :)

**Comment by geoand** [comment]
I am not sure about the refactor, because Quarkus's `quarkus.hibernate-orm.sql-load-script` property isn't a direct replacement for `spring.datasource.data` because it does not support the prefix (like `classpath:` or `file:`), so we can't just tell users to replace the property. 
We should probably just use the term "adapt" in this case so the user knows it's not a straight replace.

**Comment by geoand** [comment]
Can we also link to the issue that sparked this PR?

File: extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java (hunk 1 above)
Comment (aureamunoz) [reply extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/SpringDataJPAProcessor.java:3]: I did it manually :woman_facepalming:

**Comment by aureamunoz** [comment]
I tried a more adapted message @geoand

**Review by geoand — APPROVED** [review]
Nice, thanks!

**Comment by geoand** [comment]
@aureamunoz please squash the 2 commits. Thanks

**Comment by aureamunoz** [comment]
Done!

**Pull request merged**